src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -388,34 +455,39 @@ private HashMap<String, Object> populateReportedSectionOfShadow(Map<String, Obje
         return reported;
     }
 
-    protected void shadowUpdated(Map<String, Object> desired, Map<String, Object> reported, Integer version) {
+    protected void shadowUpdated(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 461-565 into a separate method.